Analysis/DEEP2_R23_O32/Plotting/te_metal_plots.py

```diff
@@ -384,31 +434,24 @@ def Bian_comparison(fitspath):
     bO32_DEEP = np.zeros(len(der_R23))
     bO32_MACT = np.zeros(len(der_R23_MACT))
 
-    jR23_det = np.zeros(len(metal_det))
-    jO32_det = np.zeros(len(metal_det))
     A_comparison = np.zeros(len(metal_det))
     AO_comparison = np.zeros(len(metal_det))
+
+    jR23_det = bian_calibration_R23(metal_det)
+    jO32_det = bian_calibation_O32(metal_det)
     for ii in range(len(metal_det)):
-        jR23_det[ii] = 138.0430 - 54.8284 * metal_det[ii] + 7.2954*metal_det[ii]*metal_det[ii]\
-                       - 0.32293 * metal_det[ii] * metal_det[ii] * metal_det[ii]
         A_comparison[ii] = (jR23_det[ii] - lR23[ii])
-        jO32_det[ii] = (-1/0.59) * (metal_det[ii] - 8.54)
         AO_comparison[ii] = (jO32_det[ii] - lO32[ii])
-    #for aa in range(len(metal_rl)):
-        #jR23_rl[aa] = a +b*metal_rl[aa]+c*(metal_rl[aa]*metal_rl[aa]) - d*(e+metal_rl[aa])*+rlO32[aa]
-
 
+    bR23_DEEP = bian_calibration_R23(der_OH)
+    bO32_DEEP = bian_calibration_O32(der_OH)
+    bR23_MACT = bian_calibration_R23(der_OH_MACT)
+    bO32_MACT = bian_calibration_O32(der_OH_MACT)
     for bb in range(len(bR23_DEEP)):
```

Comment:
You can also refactor out `B_comparison`, etc.  No for loop should be needed here.

Reply:
fixed